gammapy/datasets/map.py

```diff
@@ -381,6 +381,12 @@ def energy_range_fit(self):
         """Energy range maps defined by the mask_fit only."""
         return self._energy_range(self.mask_fit)
 
+    @property
+    def energy_range_total(self):
```

Comment:
I think this method won't work as soon as you have a `nan` in a mask. How does it work on full `MapDataset`?
Maybe move it to an internal function?

Reply:
@luca-giunti Could implement this change to handle NaN values?